### PR TITLE
MetaService Fixes

### PR DESCRIPTION
**Pull request by xiaoxichen**

### CODE REVIEW

File: src/tests/test_meta_blk_mgr.cpp [hunk 1]
```diff
@@ -497,13 +497,16 @@ class VMetaBlkMgrTest : public ::testing::Test {
...
         // restart will cause recovery and callbacks will be triggered
+        ++m_restart_cnt;
         m_cb_blks.clear();
         restart_homestore();
     }
 
     void validate() {
         // verify received blks via callbaks are all good;
         verify_cb_blks();
+        m_write_sbs.swap(m_cb_blks);
```
Comment (xiaoxichen) [inline src/tests/test_meta_blk_mgr.cpp:508]: use `m_cb_blks` to replace `m_write_sbs`, as all the cookies saved in `m_write_sbs` is invalid after restart

File: src/tests/test_meta_blk_mgr.cpp [hunk 2]
```diff
@@ -579,24 +582,28 @@ class VMetaBlkMgrTest : public ::testing::Test {
...
-                    m_cb_blks[mblk->hdr.h.bid.to_integer()] = md5_sum(r_cast< const char* >(buf.bytes()), size);
+                    m_cb_blks[mblk->hdr.h.bid.to_integer()] = sb_info_t {
+                        mblk, md5_sum(r_cast< const char* >(buf.bytes()), size)};
                 }
             },
-            [this](bool success) { HS_DBG_ASSERT_EQ(success, true); });
+            [this](bool success) {
+                HS_DBG_ASSERT_EQ(success, true);
+                m_total_wrt_sz = m_mbm->used_size();
+                HS_REL_ASSERT_EQ(m_mbm->total_size() - m_total_wrt_sz, m_mbm->available_blks() * m_mbm->block_size());
```
Comment (xiaoxichen) [inline src/tests/test_meta_blk_mgr.cpp:605]: update the `m_total_wrt_sz` at the end of recovery.   it was set to 0 in L590 before MetaService recovered.

**Comment by xiaoxichen** [comment]
This PR fixes https://github.com/eBay/HomeStore/issues/365

**Review by zhiteng — APPROVED** [review]
lgtm

**Review by JacksonYao287 — DISMISSED** [review]
LGTM

**Comment by JacksonYao287** [comment]
> This PR fixes #365

#365 is caused by partial write, right?

**Comment by xiaoxichen** [comment]
> > This PR fixes #365
> 
> #365 is caused by partial write, right?

Not partial write , but pwrite error with errno=22 ,  likely due to the buffer not align

**Review by JacksonYao287 — APPROVED** [review]
LGTM

**Pull request merged**